cloud/services/resourceskus/client.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceskus
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
+	"github.com/Azure/go-autorest/autorest"
+
+	azure "sigs.k8s.io/cluster-api-provider-azure/cloud"
+)
+
+// Client wraps go-sdk
+type Client interface {
+	List(context.Context, string) (compute.ResourceSkusResultPage, error)
+	HasAcceleratedNetworking(context.Context, string) (bool, error)
+}
+
+// AzureClient contains the Azure go-sdk Client
+type AzureClient struct {
+	skus compute.ResourceSkusClient
+}
+
+var _ Client = &AzureClient{}
+
+// NewClient creates a new Resource SKUs client from subscription ID.
+func NewClient(subscriptionID string, authorizer autorest.Authorizer) *AzureClient {
+	return &AzureClient{
+		skus: newResourceSkusClient(subscriptionID, authorizer),
+	}
+}
+
+// newResourceSkusClient creates a new Resource SKUs client from subscription ID.
+func newResourceSkusClient(subscriptionID string, authorizer autorest.Authorizer) compute.ResourceSkusClient {
+	c := compute.NewResourceSkusClient(subscriptionID)
+	c.Authorizer = authorizer
+	_ = c.AddToUserAgent(azure.UserAgent) // intentionally ignore error as it doesn't matter
+	return c
+}
+
+// List returns all Resource SKUs available to the subscription.
+func (ac *AzureClient) List(ctx context.Context, filter string) (compute.ResourceSkusResultPage, error) {
+	// TODO: cache these API call results?
```

Comment:
unless you plan on addressing this TODO in this PR, please move it to an issue and reference it in this PR. Trying to avoid adding code comment TODOs in the codebase in favor of GH issues.